cms/static/js/views/modals/edit_xblock.js

```diff
@@ -11,7 +11,8 @@ define(['jquery', 'underscore', 'backbone', 'gettext', 'js/views/modals/base_mod
         var EditXBlockModal = BaseModal.extend({
             events: _.extend({}, BaseModal.prototype.events, {
                 'click .action-save': 'save',
-                'click .action-modes a': 'changeMode'
+                'click .action-modes a': 'changeMode',
+                'click .title-edit-button': 'clickTitleButton'
```

Comment:
Not in this file, but for Change 3 I would add margin-left: $baseline/2 (i.e. 10px) to .title-edit-button css.